cumulusci/tasks/bulkdata/extract.py

```diff
@@ -220,27 +242,31 @@ def strip_name_field(record):
             )
         else:
             # If using the autogenerated id field, split out the returned records
-            # into two separate files and load into the main table and the sf_id_table
-
-            with tempfile.TemporaryFile("w+", newline="", encoding="utf-8") as f_values:
-                with tempfile.TemporaryFile(
-                    "w+", newline="", encoding="utf-8"
-                ) as f_ids:
-                    data_file_values, data_file_ids = self._split_batch_csv(
-                        record_iterator, f_values, f_ids
-                    )
-                    self._sql_bulk_insert_from_records(
-                        connection=conn,
-                        table=mapping["table"],
-                        columns=columns,
-                        record_iterable=csv.reader(data_file_values),
-                    )
-                    self._sql_bulk_insert_from_records(
-                        connection=conn,
-                        table=mapping["sf_id_table"],
-                        columns=["sf_id"],
-                        record_iterable=csv.reader(data_file_ids),
-                    )
+            # into two separate streams and load into the main table and the sf_id_table
+            values, ids = itertools.tee(record_iterator)
```

Comment:
Isn't `itertools.tee` very memory-expensive when you have to consume the entirety of one stream prior to the other?

Reply:
Oh, I see you did something clever to avoid that. Apologies :)

Reply:
It certainly is something to be careful with!